code/modules/mob/living/simple_animal/bot/bot.dm

```diff
@@ -227,6 +237,40 @@
 				user << "<span class='warning'>Please close the access panel before locking it.</span>"
 			else
 				user << "<span class='warning'>Access denied.</span>"
+	else if(istype(W, /obj/item/device/paicard))
+		if(paicard)
+			user << "<span class='warning'>A [paicard] is already inserted!</span>"
+		else if(allow_pai && !key)
+			if(!locked && !open)
+				var/obj/item/device/paicard/card = W
+				if(card.pai && card.pai.mind)
+					if(!user.drop_item())
+						return
+					W.forceMove(src)
+					paicard = card
+					user.visible_message("[user] inserts [W] into [src]!","<span class='notice'>You insert [W] into [src].</span>")
+					paicard.pai.mind.transfer_to(src)
+					src << "<span class='notice'>You sense your form change as you are uploaded into [src].</span>"
+					bot_name = name
+					name = paicard.pai.name
+					pai_history += text("\[[time_stamp()]\] [paicard.pai][(ismob(paicard.pai) && paicard.pai.ckey) ? "([paicard.pai.ckey])" : ""] added to [src.bot_name] by [user][(ismob(user) && user.ckey) ? "([user.ckey])" : ""]")

```

Comment:
You can replace those ugly ternaries with `key_name(paicard.pai)` and `key_name(user)`. You don't need to check if the former exists because this whole block is inside an if for it.